pep-0660.rst

```diff
@@ -189,12 +195,44 @@ cached nor distributed.
 Rejected ideas
 ==============
 
+``editable`` local version identifier
+-------------------------------------
+
 The ideas of having build backends append or modify the local version
 identifier to include the ``editable`` string has been rejected because it
 would not satisfy ``==`` version speicifier that include the local version
 identifier. In other workds ``pkg==1.0+local`` is not satisfied by version
 ``1.0+local.editable``.
 
+Virtual wheel
+-------------
+
+`Another approach was proposed <https://github.com/pypa/pip/pull/8215>`_, where
+the build backend returns a mapping from source files and directories to the
+installed layout. It is then up to the installer frontend to realize the
+editable installation by whatever mean it deems adequate for its users.
+
+In terms of capabilities, both proposals seem equivalent, at least in
```

Comment:
If you're going with this bold statement do clarify how you're planning to support headers, data files and scripts.

Reply:
Why should I? The whole point is that we don't know (in either proposal) and can let backend authors experiment. 

Regarding headers, for instance, assuming there is a use case to install them in editable mode, one can use symlinks. But is there such a use case ? This PEP explicitly says only python files must be "edtiables" and changes to other file may require an additional installation step.

Reply:
What about scripts? Those can be python files.

> Regarding headers, for instance, assuming there is a use case to install them in editable mode, one can use symlinks. But is there such a use case?

Check astropy project that ships headers. The wheel spec allows it so there is a such use case.

> Why should I? The whole point is that we don't know (in either proposal) and can let backend authors experiment.

We should at least hypothetically provide a way to achieve this, just to prove that the PEP allows that functionality.

Reply:
https://github.com/pypa/pip/pull/8215 doesn't seem related to this PEP.

We also rejected my prototype that uses the wheel structure but skips creating an archive. The archive is easy to produce; no one has to worry about cleaning a build directory (full of the unpacked wheel); we avoid describing exactly what an unpacked wheel is.

I think packages that link against other packages that provide headers are used to being able to find those headers in source checkouts, as well as in the installed location? I don't think the way installed headers work in Python is particularly great.

Most scripts are entry_points scripts handled by the standard wheel install.

`setup.py develop` is about making your code importable by other code, and providing your metadata so that your editable distribution is "installed" according to "pip list".

But old-fashioned scripts are standalone. They need to import library code, but almost never need to be "imported by" dependents. There is less pressure to copy them to the central scripts location during an editable install. If you really are editing, you may run those scripts directly from the checkout. They will work fine.

Data files are almost always relative to the source code (part of purelib/platlib) since unfortunately the data_files feature has been broken for a long time. It does not install into a predictable location given varying virtualenv / --user / global install targets.

This is a long way to say that we think the most important use case is covered without worrying about scripts, headers, and data. It's hard to manually do an in-place build with correct .dist-info metadata on the .dist-info search path. `setup.py develop` or a replacement editable system solves that.

Could line 218 be reworded to say something more like "both proposals could provide the core 'editable' feature".

My biggest concern with the virtual wheel proposal is that build systems do not work by generating the necessary mapping as an intermediate step. Providing it to the hook could be burdensome and restrict the design of 'editable-capable' build systems. And that while it's pretty obvious to see how the simple case would work, it would still be the installer's responsibility to make sense of whatever crazy mapping was provided to it by the backend - if the structure of the mapping did not resemble the structure of what was on disk?

In this PEP and in `setup.py develop` the publisher has an incentive to lay out their source code in a way that would (mostly) work with a normal `.pth` file.